src/transformers/generation_tf_utils.py

```diff
@@ -2250,30 +2245,32 @@ def sample(
         decoder_hidden_states = [] if (return_dict_in_generate and output_hidden_states) else None
 
         # 3. init tensors to use for "xla-compileable" generate function
-        # define bsz, seq_length
         batch_size, cur_len = input_ids.shape
 
-        # initialize `generated`, `finished_sequences`
+        # initialize `generated` (pre-populated with `pad_token_id`), `finished_sequences`
         generated = tf.TensorArray(
             element_shape=(batch_size,),
             dtype=tf.int32,
             dynamic_size=False,
             size=max_length,
             clear_after_read=False,
         )
-        finished_sequences = tf.zeros((batch_size,), dtype=tf.bool)
+        if pad_token_id:  # ignores the cases when it is 0 or None
+            for i in range(max_length):
+                generated = generated.write(i, tf.broadcast_to(pad_token_id, (batch_size,)))
```

Comment:
Same here, this looks like `generated` will just be a read-only array of `pad_token_id` after this!

Reply:
Ignore this - resolved as per the conversation above.

Reply:
(see comment above :P)